app/javascript/components/notification-drawer/notification-drawer.jsx

```diff
@@ -0,0 +1,188 @@
+import React, { useState } from 'react';
+import moment from 'moment';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  Button, NotificationDrawer as Drawer, Notification, MenuItem, NotificationContent, NotificationInfo, NotificationMessage, Icon,
+} from 'patternfly-react';
+import classnames from 'classnames';
+import { getNotficationStatusIconName, unreadCountText, viewDetails } from './helpers';
+import { maxNotifications as maxNotificationsConstant } from '../../packs/notification-drawer-common';
+import {
+  toggleDrawerVisibility, markNotificationRead, markAllRead, clearNotification, clearAll, toggleMaxNotifications,
+} from '../../miq-redux/actions/notifications-actions';
+
+const NotificationDrawer = () => {
+  const dispatch = useDispatch();
+  const isDrawerVisible = useSelector(({ notificationReducer: { isDrawerVisible } }) => isDrawerVisible);
+  const drawerTitle = __('Notifications');
+  const [isDrawerExpanded, setDrawerExpanded] = useState(false);
+  const [isPanelExpanded, setPanelExpanded] = useState(true);
+  const unreadCount = useSelector(({ notificationReducer: { unreadCount } }) => unreadCount);
+  const notifications = useSelector(({ notificationReducer: { notifications } }) => notifications);
+  const totalNotificationsCount = useSelector(({ notificationReducer: { totalNotificationsCount } }) => totalNotificationsCount);
```

Comment:
`useSelector` can be used for multiple values. You don't need to call the hook each time you need something from the state.

```jsx
const { isDrawerVisible, unreadCount } = useSelector(({ notificationReducer: { isDrawerVisible, unreadCount } }) => ({ isDrawerVisible, unreadCount )});
```

All state variables should probably use only one useSelector if possible.